Movement.h

```diff
@@ -18,24 +18,30 @@ class Movement {
     Movement();
     Control control;
     SensorMap sensor;
-    BNO logic;
+    BNO bno_;
 
-    // Move the robot forward.   
-    void advance(const double desire, const double desireUltrasonic);
+    // Move the robot forward with P.   
+    void advanceP(const double desire);
+    // Move the robot backwards with P.
+    void moveBackP(const double desire);
     // Turn the robot 90 degrees to the left.
-    void turnLeft(const uint8_t vel);
+    void turnLeft(const uint8_t speed);
     // Turn the robot 90 degrees to the right.
-    void turnRight(const uint8_t vel);
+    void turnRight(const uint8_t speed);
     // Turn left or right depending on the desired angle.
     void turnDegrees(double desire);
     // Stop the robot.
     void stopEngines();
     // Go fast to the ramp.
     void fastForward();
+    // Go back fast.
+    void fastBackward();
     // Move the robot backwards.
     void moveBack();
-    // Move the robot forward through the pwm asigned in each side.
+    // Move the robot forward through the pwm assigned in each side.
     void forwardPwm(const uint8_t pwm_right, const uint8_t pwm_left);
+    // Move the robot backward through the pwm assigned in each side.
+    void backwardPwm(const uint8_t pwm_right, const uint8_t pwm_left);
```

Comment:
You can have one Motors.cpp and .h file, where all the functions are literally these ones + the setup one. Functions where the only thing you do is send the velocity or stop them and that's it, nothing of calculations.